acoustid/scripts/claim_duplicate_gids.py

```
@@ -0,0 +1,266 @@
+        # the snapshot with identical content. Either way the group has its
+        # primary and moving the gid would invalidate what is grouped on it.
+        "    AND NOT EXISTS (SELECT 1 FROM meta m2 WHERE m2.gid = c.gid)".format(
+            table=check_table_name(dups_table)
+        )
+    )
+    claimed = fingerprint_db.execute(query, {"lo": str(lo), "hi": str(hi)}).rowcount
+    fingerprint_db.execute(
+        sql.text(
+            "UPDATE {t} SET last_gid = CAST(:hi AS uuid),"
+            " groups_claimed = groups_claimed + :claimed, updated = now()"
+            " WHERE id = 1".format(t=PROGRESS_TABLE)
```

Ruff (0.16.3)

[error] Possible SQL injection vector through string-based query construction

(S608)